docs/book/src/developer/architecture/controllers/control-plane.md

```diff
@@ -110,6 +112,10 @@ documentation][scale].
   deletion. A duration of 0 will retry deletion indefinitely. It defaults to 10 seconds on the
   Machine.
 
+#### Optional `spec` fields for implementations providing endpoints
+
+The `ImplementationControlPlane` object may provide a `spec.controlPlaneEndpoint` field to inform the Cluster controller where the endpoint is located.
```

Comment:
Let's also define the type of the field please (I assume it's the same as in InfraCluster, but let's be explicit)